schemaregistry/rest_service.go

```diff
@@ -133,13 +133,18 @@ func newRestService(conf *Config) (*restService, error) {
 
 	timeout := conf.RequestTimeoutMs
 
+	client := conf.BaseClient
+	if client == nil {
+		client = new(http.Client)
+	}
+
+	client.Transport = transport
+	client.Timeout = time.Duration(timeout) * time.Millisecond
```

Comment:
I wonder if the "old" config should be deprecated. Also, it might make sense to check if the client has any existing timeout and use that instead if so.